lib/agent/https_proxy_ocsp_agent.js

```diff
@@ -1,278 +1,33 @@
-
-  // if `true`, then connect to the proxy server over TLS. defaults to `false`.
-  HttpsAgent.secureProxy = proxy.protocol ? /^https:?$/i.test(proxy.protocol) : false;
-
-  // prefer `hostname` over `host`, and set the `port` if needed
-  proxy.host = proxy.hostname || proxy.host;
-  proxy.port = +proxy.port || (this.secureProxy ? 443 : 80);
-
-  if (proxy.host && proxy.path)
-  {
-    // if both a `host` and `path` are specified then it's most likely the
-    // result of a `url.parse()` call... we need to remove the `path` portion so
-    // that `net.connect()` doesn't attempt to open that as a unix socket file.
-    delete proxy.path;
-    delete proxy.pathname;
-  }
-
-  if (proxy.user && proxy.password)
-  {
-    // user:password
-    proxy.auth = proxy.user + ':' + proxy.password;
-    delete proxy.user;
-    delete proxy.password;
-
-    if (!HttpsAgent.secureProxy)
-    {
-      console.warn("Warning: connecting to an authenticated proxy server through HTTP. To use HTTPS, set 'proxyProtocol' to 'HTTPS'")
-    }
-  }
+module.exports = createHttpsProxyAgent;
 
-  HttpsAgent.proxy = proxy;
-  return HttpsAgent;
+function createHttpsProxyAgent(opts) {
+  // HttpsProxyAgent >= 6.x takes two arguments for its constructor
+  // See: https://github.com/TooTallNate/proxy-agents/blob/main/packages/https-proxy-agent/CHANGELOG.md#600
+  const { host: hostname, port, user: username, password, protocol: rawProtocol, ...agentOptions } = opts;
+  let protocol = rawProtocol.endsWith(':') ? rawProtocol : `${rawProtocol}:`;
+  return new HttpsProxyOcspAgent({ hostname, port, username, password, protocol }, agentOptions);
 }
 
-inherits(HttpsProxyAgent, createAgent);
-
-/**
- * Called when the node-core HTTP client library is creating a new HTTP request.
- *
- * @api public
- */
-
-function connect(req, opts, fn)
-{
-  var proxy = this.proxy;
-  var agent = this;
-  Logger.getInstance().debug("Using proxy=%s for host %s", proxy.host, opts.host);
-
-  // create a socket connection to the proxy server
-  var socket;
-  if (this.secureProxy)
-  {
-    socket = tls.connect(proxy);
+class HttpsProxyOcspAgent extends HttpsProxyAgent {
+  constructor(proxy, opts) {
+    super(proxy, opts);
   }
-  else
-  {
-    socket = net.connect(proxy);
-  }
-
-  // we need to buffer any HTTP traffic that happens with the proxy before we get
-  // the CONNECT response, so that if the response is anything other than an "200"
-  // response code, then we can re-play the "data" events on the socket once the
-  // HTTP parser is hooked up...
-  var buffers = [];
-  var buffersLength = 0;
 
-  function read()
-  {
-    var b = socket.read();
-    if (b)
-    {
-      ondata(b);
+  async connect(req, opts) {
+    Logger.getInstance().debug("Using proxy=%s for host %s", this.proxy.hostname, opts.host);
+    const socket = await super.connect(req, opts);
+    if (socket instanceof tls.TLSSocket) {
+      SocketUtil.secureSocket(socket, opts.host, this)
     }
```

Comment:
The code is much less verbose, but it does the same thing as before:

1. Wait for the socket to be upgraded
2. Call `SocketUtil.secureSocket()` to block any requests until OCSP validation has succeeded

The best way I found to verify this works as expected is by enabling debug logging so you can see the OCSP requests/etc.